src/http_reader.rs

```diff
@@ -72,30 +75,25 @@ fn get_data(filename: &str, range: Vec<ByteRangeSpec>) -> Result<ResponseData, S
     })
 }
 
-fn get_content_length(response: &reqwest::Response) -> Option<u64> {
-    match response.headers().get::<ContentRange>() {
-        Some(&ContentRange(ContentRangeSpec::Bytes {
-            range: _range,
-            instance_length,
-        })) => instance_length,
-        _ => match response.headers().get::<ContentLength>() {
-            Some(length) => Some(**length as u64),
-            None => None,
-        },
-    }
-}
-
 fn get_content_range(response: &reqwest::Response) -> Result<Option<u64>, String> {
-    match response.headers().get::<ContentRange>() {
-        Some(content_range) => match content_range.clone() {
-            ContentRange(ContentRangeSpec::Bytes {
-                range: _range,
-                instance_length: length,
-            }) => Ok(length),
-            ContentRange(ContentRangeSpec::Unregistered {
-                unit: _unit,
-                resp: _resp,
-            }) => Err("Unregistered, actually unsupported".to_string()),
+    match response.headers().get("Content-Range") {
```

Comment:
it can be updated more like that (with more rust spirit):
```rust
fn get_content_range(response: &reqwest::Response) -> Result<Option<u64>, String> {
    if let Some(content_range) = response.headers().get(reqwest::header::CONTENT_RANGE) {
        let content_range_str = content_range
            .to_str()
            .map_err(|msg| format!("Inable serializing header value to str: {}", msg))?;

        match ContentRangeSpec::from_str(content_range_str)
            .map_err(|msg| format!("Error parsing content range from str: {}", msg))?
        {
            ContentRangeSpec::Bytes {
                instance_length: length,
                ..
            } => Ok(length),
            ContentRangeSpec::Unregistered { .. } => {
                Err("Unregistered range, actually unsupported".to_string())
            }
        }
    } else {
        Err("Missing content_range".to_string())
    }
}
```